streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -862,6 +871,31 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
         return props;
     }
 
+    /**
+     * Get a map of custom configs by removing from the originals all the Streams, Consumer, Producer, and AdminClient configs.
+     * Prefixed properties are also removed because they are already added by {@link #getClientPropsWithPrefix(String, Set)}.
+     * This allows to set a custom property for consumer or producer alone if specified using a prefix, or for both
+     * when no prefix is used.
+     *
+     * @param consumerPrefix the prefix used to denote Consumer-only properties
+     * @param producerPrefix the prefix used to denote Producer-only properties
+     * @param adminClientPrefix the prefix used to denote AdminClient-only properties
+     * @return a map with the custom properties
+     */
+    private Map<String, Object> getClientCustomProps(final String consumerPrefix,
```

Comment:
Do we need to pass in these three parameters if they are supposed to never change and all its callers are passing the same values?

Reply:
Thanks for the review! Yes, definitely unnecessary. I'm removing the arguments. Additionally, since we're just adding the custom properties to the  props map, I'm moving the call to `props.putAll(getClientCustomProps())` instead assigning it to a variable.